dask/array/rechunk.py

```diff
@@ -221,7 +221,7 @@ def rechunk(x, chunks="auto", threshold=None, block_size_limit=None):
     >>> y = x.rechunk({0: -1, 1: 'auto'}, block_size_limit=1e8)
     """
     # don't rechunk if array is empty
-    if x.ndim > 0 and x.shape[-1] == 0:
+    if x.ndim > 0 and not [s for s in x.shape]:
```

Comment:
I find this a little more clear:

```suggestion
    if x.ndim > 0 and all(s == 0 for s in x.shape):
```